tachyon/zk/base/commitments/shplonk_extension.h

```diff
@@ -151,8 +151,8 @@ class SHPlonkExtension final
   }
 
  private:
-  friend class c::zk::plonk::halo2::KZGFamilyProverImpl<
-      SHPlonkExtension<Curve, MaxDegree, MaxExtendedDegree, Commitment>>;
+  template <typename PCS, typename LS>
+  friend class c::zk::plonk::halo2::KZGFamilyProverImpl;
```

Comment:
I'm not sure but this doesn't work right?
```suggestion
  friend class c::zk::plonk::halo2::KZGFamilyProverImpl<PCS, LS>;
```

Reply:
No, it doesn't work because `PCS` and `LS` didn't declared in this scope!